.github/workflows/asv_check.yml

```diff
@@ -0,0 +1,34 @@
+name: asv
+
+# CI ASV CHECK is aimed to verify that the benchmarks execute without error.
+on: [pull_request, push]
+
+jobs:
+  quick:
+    runs-on: ubuntu-latest
+    defaults: 
+      run:
+        shell: bash -el {0}
+
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+
+      - name: Install Python
+        uses: actions/setup-python@v3
+        with:
+          python-version: '3.9.7'
+
+      - name: Install asv
+        run: pip install asv==0.4.2
+
+      - name: Run asv benchmarks
+        run: |
+          cd benchmarks
+          asv machine --yes
+          asv run --quick --dry-run --show-stderr | sed "/failed$/ s/^/##[error]/" | tee benchmarks.log
```

Comment:
Ah, it's talking about `master` because that's the default if you don't specify a specific commit or commit range.  Try using `HEAD^!` here (`asv run HEAD^! --quick ...`), which I think will tell asv to just run the currently checked-out commit.

Reply:
Yes, I think it's the key. I was always thinking about the `asv.conf` before this. 
Thanks for your valuable advice!